### DISCUSSION TITLE
Determine Chirality

### DISCUSSION THREAD

**Discussion by Old-Shatterhand** [rdkit/rdkit#5427] [Q&A] [answered]
Hi community,

I have a problem determining the chirality of atoms in RDKit molecules. I uploaded a notebook with examples to [colab](https://colab.research.google.com/drive/1S1G3LGn-rKNmgwEk4_qhbHjOQFhYQb1p?usp=sharing).

There is the [Natta projection](https://en.wikipedia.org/wiki/Natta_projection) (in german: [Keilstrichform](https://de.wikipedia.org/wiki/Keilstrichformel)) that is used by RDKit to visualize molecules (as you can see in the Notebook). I need to have a function that determines if the bond is light or dark given the ring atoms. 
Actually, I'm interested in the orientation of the chiral carbon atoms, i.e. is the sidechain above or below the ring. The function should be invariant to different SMILES of the same molecule and report a mapping of atom-indices to orientation specifiers.

Unfortunately, I cannot use the `atom.GetChiralTag()` as they depend on the input SMILES and change with the input order of the atoms.
Also, I cannot use `FindMolChiralCenters(mol1)` (also not in combination with `AssignStereochemistry(mol3, cleanIt=True, force=True, flagPossibleStereoCenters=True)`) as shown in the notebook, because the `R` and `S` tags seem to be independent of the light and dark bonds in the visualization.

Basically, I see two options to solve my problem:
 1. Get a function that computes what I described above or
 2. Find a way to tell RDKit to generate a SMILES string for the molecules by considering the ring-atoms as the backbone starting in the ring's oxygen atom. Then, the ring atoms have the same IDs (if both molecules are six-rings) and the chirality should be the same in mol2 and mol3 from the notebook. Here, I need an additional mapping of the input-SMILES to the reordered SMILES.

Thank you for your help.
Roman

**Comment by greglandrum** [comment, marked as answer]
Hi @Old-Shatterhand: It sounds like you want to know which bonds are drawn as wedges. You can do that by asking the RDKit to wedge the bonds and then querying them. I've made some changes to your colab notebook to show how to do this:
https://colab.research.google.com/drive/14SeUdtp_mjKH8KeIiH58XTczhqVUkswt?usp=sharing

Before you over-interpret these, it's important to realize that there are multiple possible ways of drawing the same molecule. So, for example, these two molecules are exactly the same thing even though the bonds are drawn with different wedges:
![image](https://user-images.githubusercontent.com/540511/180926295-7f63ab62-8409-4b23-b00d-671eed4deb32.png)
and
![image](https://user-images.githubusercontent.com/540511/180926342-202a70c0-6425-411a-ae54-47782ae4bec0.png)

What exactly are you trying to do?

**Reply by Old-Shatterhand** [reply]
Hi @greglandrum,
thank you very much for your answer. It's part of my work on the [GlyLES package](https://pypi.org/project/glyles/). You might have seen it at the GCC 2022, I had a poster there.

At some point, I need to know which atoms of a molecule are added and which are part of the original molecule. So, when I have galactose and 3-sulfo-galactose, I want to know which atoms of 3-sulfo-galactose are added by some attached functional groups and which atoms are also part of galactose. This plays a role when enumerating carbon atoms in sugars with added functional groups with carbon atoms, e.g. a benzene ring. 
My plan was to use some graph isomorphism methodology for this and I thought I have to do it chirality sensitive. I figured out that that's not necessary or I haven't found the edge case where chirality in the isomorphism is curcial ;-).

Now, that I know that chirality in rdkit is not as simple as I thought, I might have to rework it based on your answer. Thank you again.
